mlir/include/mlir/Dialect/MIOpen/MIOpenOps.td

```diff
@@ -481,10 +481,10 @@ def MIOpen_ThreadwiseGemmOp:
 def MIOpen_MFMAV2Op:
     MIOpen_Op<"mfma_v2", [AllTypesMatch<["sourceA", "sourceB"]>,
       AllTypesMatch<["destC", "destD"]>]>,
-    Arguments<(ins AnyTypeOf<[F32, VectorOfLengthAndType<[4], [F16]>, VectorOfLengthAndType<[2], [I16]>]>: $sourceA,
-                   AnyTypeOf<[F32, VectorOfLengthAndType<[4], [F16]>, VectorOfLengthAndType<[2], [I16]>]>: $sourceB,
-                   VectorOfRankAndType<[1], [F32, F16]>: $destC)>,
-    Results<(outs VectorOfRankAndType<[1], [F32, F16]>: $destD)> {
+    Arguments<(ins AnyTypeOf<[I32, F32, VectorOfLengthAndType<[4], [F16]>, VectorOfLengthAndType<[2], [I16]>]>: $sourceA,
+                   AnyTypeOf<[I32, F32, VectorOfLengthAndType<[4], [F16]>, VectorOfLengthAndType<[2], [I16]>]>: $sourceB,
```

Comment:
Do we instead want to call this `vector<4xi8>` since that's what it is? `bitcast` exists on the way down?

Or will that transition live elsewhere in the pipeline?

Reply:
From my limited knowledge in those passes, I can see miopen->gpu->xdlops are mostly a smooth 1:1 transition, so it doesn't look like the I32 <-> vector<4xi8> transition happens in those last couple of passes. 

@whchung Which pass do you think should we do the packing?